### ISSUE TITLE
plz add support for MX35LF4G24AD-Z4I

### ISSUE THREAD

**Issue by li60240** [Droid-MAX/SNANDer#17]
MX35LF4G24AD
MX35LF4G24AD-Z4I

**Comment by Droid-MAX** [comment]
Well done, feedback in time plz

**Comment by li60240** [comment]
thanks for your hard work.I have try flash the spinand with on die ecc,buy something wrong.
status code:Bad in the end.
![mmexport1719159007613](https://github.com/Droid-MAX/SNANDer/assets/15099072/9428756a-dc8d-4370-b597-f43cfd672b54)

**Comment by li60240** [comment]
> thanks for your hard work.I have try flash the spinand with on die ecc,buy something wrong. status code:Bad in the end. ![mmexport1719159007613](https://private-user-images.githubusercontent.com/15099072/342085935-9428756a-dc8d-4370-b597-f43cfd672b54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxOTM0NjksIm5iZiI6MTcxOTE5MzE2OSwicGF0aCI6Ii8xNTA5OTA3Mi8zNDIwODU5MzUtOTQyODc1NmEtZGM4ZC00MzcwLWI1OTctZjQzY2ZkNjcyYjU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI0VDAxMzkyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxZjAzYjUyNzk0YzI4ZGZkNmRkOGE5OTE2YzVjYTEwMGU2OTJmZjcxZDYwNTZlYTIzMjJlMTVkZmQzNmVkY2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KhP8jmvOhV-aSoQJ3fj_Xgqs3_wWqVL5AJkdPZHtX4U)

disable ecc flashing it also failed with code Bad.
![1635409b71ead58b000e27615e6a51a](https://github.com/Droid-MAX/SNANDer/assets/15099072/23c61070-95be-4f90-8bff-fafcc11635df)

**Comment by Droid-MAX** [comment]
Something's wrong. It only took 26 seconds to write？

**Comment by li60240** [comment]
> Something's wrong. It only took 26 seconds to write？

是的，我只刷写了5M文件，开启ecc或者关闭ec，验证全部识别

**Comment by Droid-MAX** [comment]
你没先给擦除一下？

**Comment by li60240** [comment]
> 你没先给擦除一下？

刷写之前有使用 -e选项先擦除的。有尝试过刷写整个带oob数据的固件（544M）。
流程如下:1 先使用命令 snander -d -e 擦除芯片。
2. 使用命令 snander -d -v -w.544.bin 刷写。
3. 结果验证环节status code都显示BAD

**Comment by li60240** [comment]
> 你没先给擦除一下？

查看了下这芯片的datasheet，MX35LF2G(4G)24AD-Z4I版本是two-plane双层结构，必须开启ecc,并且需要Plane Select 位，这个可以
![S40625-22134658_cn wps moffice lite meizu](https://github.com/Droid-MAX/SNANDer/assets/15099072/8c576d6f-c7d1-4c9c-8556-332e18e20e7e)
适配吗

**Comment by Droid-MAX** [comment]
now you can download latest release and try again.

**Comment by li60240** [comment]
> now you can download latest release and try again.
I have downloaded the latest version and tried again. The same problem still exists with chip MX35LF4G24AD-Z4I and it cannot pass the verification process.Same error as the previous version.
![20240626135233](https://github.com/Droid-MAX/SNANDer/assets/15099072/c60de4ac-6b4f-4ad5-8aa8-82e212b0fd80)
![20240626135242](https://github.com/Droid-MAX/SNANDer/assets/15099072/bfa89d49-e4d8-4bce-ba77-0f23d8c66b5f)

**Comment by Droid-MAX** [comment]
Can test with disable internal ecc features?

**Comment by li60240** [comment]
> Can test with disable internal ecc features?

i haved disable the ecc feature and flashing verification process failed as before.
![mmexport1719399542421](https://github.com/Droid-MAX/SNANDer/assets/15099072/ac6ed7bf-fd34-4a7f-98e2-e97c68275592)
I have already erased the chip with "-d" before starting to flash it.

**Comment by Droid-MAX** [comment]
@li60240 试过全片编程没？

**Comment by li60240** [comment]
> @li60240 试过全片编程没？

尝试过，我上面给你的反馈截图就是全片512M+32M 禁用ecc特性，全片编程。

**Comment by Droid-MAX** [comment]
@li60240 最近我关注到恩山论坛上的一些相关的帖子里也有提到过类似的问题，我想再核实确认一下你刷入的固件文件是如何备份提取出来的？以及机器用的闪存芯片是否一致？

https://www.right.com.cn/forum/thread-8420628-1-1.html

**Comment by pzliliang** [comment]
[F35SQA001G-WWT](https://item.szlcsc.com/5727685.html?fromZone=s_s__%2522F35SQA%2522&spm=sc.gb.xh1.zy.n___sc.gb.hd.ss&lcsc_vid=R1gIBgJWQwJYAVwCQFhcBVAARlhcAV1WFlQIU1FVR1kxVlNSRVBWVFBRQ1NXUDsOAxUeFF5JWBYZEEoVDQ0NFAdIFA4DSA%3D%3D)

ID: CD71


![Image](https://github.com/user-attachments/assets/6805bb05-c1c1-4d44-8700-eed2564a46e5)

**Comment by pzliliang** [comment]
F35SQA512M-WWT

ID:CD70

![Image](https://github.com/user-attachments/assets/cd60856e-7c6c-43aa-b010-216fc369ac75)

**Comment by pzliliang** [comment]
F35SQA002G-WWT

ID:CD72

![Image](https://github.com/user-attachments/assets/f2a06fb8-a8d9-47b6-ada0-51413710e953)

**Comment by pzliliang** [comment]
ZD35Q2GC-IB

ID:2C24

![Image](https://github.com/user-attachments/assets/daa45a3f-18bc-4f3b-b1e3-e58b95289ea5)

**Comment by pzliliang** [comment]
ZD35Q1GC-IB
ID:BA71

![Image](https://github.com/user-attachments/assets/43fbffa4-5b29-4af4-8bbb-a3d04fdcb48a)

![Image](https://github.com/user-attachments/assets/0de71a33-0e4a-4524-95d2-f4767a0a4cd5)

**Comment by Droid-MAX** [comment]
@pzliliang DO NOT SPAM THE ISSUES.

**Comment by pzliliang** [comment]
> [@pzliliang](https://github.com/pzliliang) DO NOT SPAM THE ISSUES.

I submitted flash data, why is it spam?

**Comment by Droid-MAX** [comment]
Please open a new issue describing your requirement or problem.
